internal/support/cronscheduler/application/scheduleactionrecord.go

```diff
@@ -109,10 +115,136 @@ func DeleteScheduleActionRecordsByAge(ctx context.Context, age int64, dic *di.Co
 	return nil
 }
 
+// GenerateMissedScheduleActionRecords generates missed schedule action records
+func GenerateMissedScheduleActionRecords(ctx context.Context, dic *di.Container, job models.ScheduleJob, latestRecords []models.ScheduleActionRecord) errors.EdgeX {
+	dbClient := container.DBClientFrom(dic.Get)
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	correlationId := correlation.FromContext(ctx)
+
+	// All latest records should have the same creation time
+	lastRecordTimestamp := latestRecords[0].Created
+
+	// Compare the last record timestamp with the job's modified timestamp to get the latest time
+	latestTime := time.UnixMilli(lastRecordTimestamp)
+	modified := time.UnixMilli(job.Modified)
+	if latestTime.Before(modified) {
+		latestTime = modified
+	}
+
+	// Generate missed runs based on the schedule type
+	missedRuns, err := generateMissedRuns(job.Definition, latestTime)
+	if err != nil {
+		lc.Errorf("Failed to generate missed records of job: %s. Correlation-ID: %s", job.Name, correlationId)
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+
+	if len(missedRuns) != 0 {
+		var missedRecordCreatedTime = time.Now().UnixMilli()
+		for _, run := range missedRuns {
+			for _, action := range job.Actions {
+				actionRecord := models.ScheduleActionRecord{
+					JobName:     job.Name,
+					Action:      action,
+					Status:      models.Missed,
+					ScheduledAt: run.UnixMilli(),
+					Created:     missedRecordCreatedTime,
+				}
+
+				record, err := dbClient.AddScheduleActionRecord(ctx, actionRecord)
```

Comment:
add a new function dbClient.AddScheduleActionRecords(ctx, []actionRecord) to improve the performance